app/abci.go

```diff
@@ -0,0 +1,160 @@
+package app
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"sort"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lazyledger/lazyledger-app/x/lazyledgerapp/types"
+	abci "github.com/lazyledger/lazyledger-core/abci/types"
+	core "github.com/lazyledger/lazyledger-core/proto/tendermint/types"
+)
+
+// This file should contain all of the althered ABCI methods
+
+// PreprocessTxs fullfills the lazyledger-core version of the ACBI interface, by
+// performing basic validation for the incoming txs, and by cleanly separating
+// share messages from transactions
+// todo(evan): refactor out a for loop.
+func (app *App) PreprocessTxs(txs abci.RequestPreprocessTxs) abci.ResponsePreprocessTxs {
+	squareSize := app.SquareSize()
+	shareCounter := uint64(0)
+	var shareMsgs []*core.Message
+	var processedTxs [][]byte
+	for _, rawTx := range txs.Txs {
+		// decode the Tx
+		tx, err := app.txConfig.TxDecoder()(rawTx)
+		if err != nil {
+			continue
+		}
+
+		// don't do anything if the transaction doesn't contain a PayForMessage sdk.Msg
+		if !hasWirePayForMessage(tx) {
+			processedTxs = append(processedTxs, rawTx)
+			continue
+		}
+
+		// only support transactions that contain a single sdk.Msg for now
+		if len(tx.GetMsgs()) != 1 {
+			continue
+		}
```

Comment:
> only support transactions that contain a single sdk.Msg for now

Is that going to change? Any other tx with multiple sdk.Messages would have already been "processed" in that `if !hasWirePayForMessage` branch I guess?

Reply:
I don't think it's going to change, but I wasn't sure and meant to ask everyone. The sdk will only commit to the state if all `sdk.Msg`s in an `sdk.Tx` are valid and successfully ran, and since we are planning to burn the fee's for `WirePayForMessage` before the rest of the `sdk.Msg`s are executed in `DeliverTx`, things could get complicated if we allow more than one `sdk.Msg` in `sdk.Tx`s that include a `WirePayForMessage`.